packages/generator/tests/acceptance/test_streaming_measurement_client.py

```diff
@@ -49,6 +51,64 @@ def test___measurement_plugin_client___stream_measure___returns_output(
     assert responses == expected_output
 
 
+def test___measurement_plugin_client___invoke_measure_from_two_threads___initiates_first_measure_and_rejects_second_measure(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    with pytest.raises(RuntimeError) as exc_info:
+        measure_thread = threading.Thread(target=measurement_plugin_client.measure)
+        measure_thread.start()
+        measurement_plugin_client.measure()  # Calling Measure in main thread, to catch error in pytest.
+        measure_thread.join()
+
+    assert "A measure call is already in progress." in exc_info.value.args[0]
+
+
+def test___non_streaming_measurement_execution___cancel___returns_true(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+    measure_thread = threading.Thread(target=measurement_plugin_client.measure)
+    measure_thread.start()
+    time.sleep(2)  # Wait for 2 seconds to call Cancel API.
```

Comment:
Hardcoded delays in tests are discouraged because they contribute to both slowness and flakiness of the test suite.

You could write a test measurement that signals an event and then waits until canceled. Then the test can wait for the event before calling cancel().

Reply:
Event is added in measurement already. This delay was given, to wait for the measure_response gRPC call to send to complete few iterations (as this is a streaming measurement) before cancelling, before cancelling. I see, that is not a best practice. So, removed the delay.